Update makefile with examples target and add versioning.

craigyam opened the pull request:
- Add test.examples makefile target
- Version the examples images

craigyam committed:
Add makefile target to run examples automated testing.

craigyam committed:
Add support for versioned examples images.

- Build the examples images with a tag based on release.
- Generate the examples yaml files.

gicaceres commented on testing/generate_example_yaml.sh:
```diff
@@ -0,0 +1,48 @@
+set -o errexit
+
+# The release for which the yamls are being generated
+TAG=$1
```
we should exit if TAG is empty

craigyam replied:
Done

gicaceres commented on testing/kubernetes/test-kubernetes.sh:
```diff
@@ -18,6 +18,7 @@ set -x
 set -o errexit
 
 A8_TEST_SUITE=$1
+A8_RELEASE=$2
```
exit if empty

craigyam replied:
Validated this in the test suite check section.

gicaceres commented on testing/docker/test-docker.sh:
```diff
@@ -18,6 +18,7 @@ set -x
 set -o errexit
 
 A8_TEST_SUITE=$1
+export A8_RELEASE=$2
```
We should exit if empty

craigyam committed:
Code review comments.

Validate the release for examples path in the test script and
the generate example yaml script.

gicaceres reviewed (approved):
LGTM

craigyam commented:
Need to update the bluemix.cfg as well.  Will generate it in the generate_example_yaml.sh script.

gicaceres reviewed (changes requested):
Need to update bluemix examples

craigyam committed:
Additional code review updates and add kibana yamls.

Clean up the example yaml files which are now generated by
make build.exampleapps (or when running the examples).
Add yamls to start the kibana containers.
Update the bluemix.cfg file with the versioned image names.

craigyam commented:
Removed the yamls from the examples directory as they are generated by testing/generate_example_yaml.sh (make build.exampleapps).  They should still be included in the zip file generated by make release with the appropriate version number.

gicaceres reviewed (approved):
LGTM

rshriram reviewed (approved):
LGTM. Have you verified that when you tag a commit in master, and run make examples, you get the example YAML files in examples/folder?

craigyam commented:
@rshriram I haven't tested it out in master, but I tested out the follow up push_dockerhub.sh changes (in a different fork) and verified that the yamls are generated with the version numbers included.
